packages/react-composites/src/composites/common/ControlBar/CustomButton.tsx

```diff
@@ -132,8 +132,18 @@ const generateCustomControlBarButtons = (
     ? customButtons
         .filter((buttonProps) => buttonProps.placement === placement)
         .map((buttonProps, i) => (internalProps) => {
-          if (buttonProps['onRenderButton']) {
-            return buttonProps['onRenderButton'](internalProps);
+          // onRenderButton is a custom prop that can be passed to override the default rendering of the button
+          // This is useful for custom buttons that need to render more than just an icon and label and is used
+          // for CallWithChat button notification badge. However, onRenderButton is not a public prop and is
+          // not documented in the API. This is a temporary solution and will need to be revisited.
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          if (buttonProps['onRenderButton' as any]) {
```

Comment:
Consider still typing instead of using any:
```
if ('onRenderButton' in buttonProps) {
  return (buttonProps['onRenderButton'] as (props: ControlBarButtonProps) => JSX.Element)(internalProps);
}
```

Reply:
(even better but more work, would be to definitely type `as (props: ControlBarButtonProps) => JSX.Element` and share that type, e.g.
```
export type _InternalCustomButtonType = (props: ControlBarButtonProps) => JSX.Element;
...
  return (buttonProps['onRenderButton'] as _InternalCustomButtonType)(internalProps);`
```

Then where it is used
```
onRenderButton: _InternalCustomButtonType = () => ...
```

Reply:
Would solution #2 add a new prop in CustomCallControlButtonProps?
If so, this would be an improvement plan and out of scope for this specific pr. I will create a separate bug for this that will require the addition of a new prop for custom button injection.

For now I will go with solution #1 and expand on it with solution 2 on a separate tracked item

Reply:
I was thinking just an internal type that only we use internally, and we just apply it here and to the one usage of onRenderButton. But its not strictly necessary as these are unlikely to change anyway or be reused elsewhere